other/dpll.py

```diff
@@ -0,0 +1,314 @@
+"""
+Davis–Putnam–Logemann–Loveland (DPLL) algorithm is a complete, backtracking-based search algorithm 
+for deciding the satisfiability of propositional logic formulae in conjunctive normal form, i.e. for solving the CNF-SAT problem.
+
+For more information about the algorithm: https://en.wikipedia.org/wiki/DPLL_algorithm
+"""
+
+import random
+
+
+class Clause():
+    """
+    A clause represented in CNF.
+    A clause is a set of literals, either complemented or otherwise.
+    For example:
+        {A1, A2, A3'} is the clause (A1 v A2 v A3')
+        {A5', A2', A1} is the clause (A5' v A2' v A1) 
+    """
+
+    def __init__(self, no_of_literals, literals):
+        """
+        Represent the number of literals, the literals themselves, and an assignment in a clause."
+        """
+        self.no_of_literals = no_of_literals
+        self.literals = [l for l in literals]
+        self.literal_values = dict()
+        for l in self.literals:
+            self.literal_values[l] = None    # Assign all literals to None initially
```

Comment:
```suggestion
                self.literals = {literal: None for literal in self.literal_values}
```
1. no_of_literals is len(self.literals)
2. The old literals is now self.literals.keys()